crypto/build.rs

```diff
@@ -3,28 +3,32 @@
 #[cfg(feature = "rustcrypto")]
 use std::ops::Deref;
 
+#[cfg(any(feature = "rustcrypto", feature = "openssl"))]
+use std::fs;
+
+use std::{env, path::Path};
+
 fn main() {
-    #[cfg(feature = "openssl")]
-    {
-        use {
-            fs::File,
-            openssl::{ec, nid, pkey},
-            std::env,
-            std::fs,
-            std::io::Write,
-            std::path::Path,
-        };
+    const ALIAS_PRIV_256: &str = "../platform/src/test_data/key_256.pem";
+    const ALIAS_PRIV_384: &str = "../platform/src/test_data/key_384.pem";
 
-        const ALIAS_PRIV_256: &str = "../platform/src/test_data/key_256.pem";
-        const ALIAS_PRIV_384: &str = "../platform/src/test_data/key_384.pem";
+    const ALIAS_PRIV_MLDSA_87: &str = "../platform/src/test_data/key_mldsa_87.pem";
 
-        const CURVE_ID_256: nid::Nid = nid::Nid::X9_62_PRIME256V1;
-        const CURVE_ID_384: nid::Nid = nid::Nid::SECP384R1;
+    println!("cargo:rerun-if-changed={ALIAS_PRIV_256}");
+    println!("cargo:rerun-if-changed={ALIAS_PRIV_384}");
+    println!("cargo:rerun-if-changed={ALIAS_PRIV_MLDSA_87}");
+
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+    let _path_256 = Path::new(&out_dir).join("alias_priv_256.pem");
+    let _path_384 = Path::new(&out_dir).join("alias_priv_384.pem");
 
-        println!("cargo:rerun-if-changed={ALIAS_PRIV_256}");
-        println!("cargo:rerun-if-changed={ALIAS_PRIV_384}");
+    //TODO(clundin): Refactor / include support for ML-DSA 87 when the openssl crate supports it.
+    #[cfg(feature = "openssl")]
+    let (_pem_256, _pem_384) = {
```

Comment:
Correct me if I'm wrong, but I think `_pem_256` and `_pem_384` can both have their leading underscores removed because they are used to write the file at the end.

Reply:
Good point! I shifted things around to remove the underscores without introducing an unused variable warning.